activerecord/lib/active_record/railtie.rb

```diff
@@ -134,23 +134,21 @@ class Railtie < Rails::Railtie # :nodoc:
             )
             next if db_config.nil?
 
+            current_version = ActiveRecord::Migrator.current_version
+            next if current_version.nil?
+
             filename = ActiveRecord::Tasks::DatabaseTasks.cache_dump_filename(
               db_config.spec_name,
               schema_cache_path: db_config.schema_cache_path,
             )
+            cache = ActiveRecord::ConnectionAdapters::SchemaCache.load_from(filename)
 
-            if File.file?(filename)
-              current_version = ActiveRecord::Migrator.current_version
-
-              next if current_version.nil?
-
-              cache = YAML.load(File.read(filename))
-              if cache.version == current_version
-                connection_pool.schema_cache = cache.dup
-              else
-                warn "Ignoring db/schema_cache.yml because it has expired. The current schema version is #{current_version}, but the one in the cache is #{cache.version}."
-              end
+            if cache.version != current_version
```

Comment:
`cache` can be `nil` now when `filename` doesn't exist, so `cache.version` can fail.

The railties tests are failing because of that.

Reply:
Ah, of course! Thank you.